src/transfer-manager.ts

```diff
@@ -0,0 +1,342 @@
+   * ```
+   * @experimental
+   */
+  async downloadManyFiles(
+    filesOrFolder: File[] | string[] | string,
+    options: DownloadManyFilesOptions = {}
+  ): Promise<void | DownloadResponse[]> {
+    const limit = pLimit(
+      options.concurrencyLimit || DEFAULT_PARALLEL_DOWNLOAD_LIMIT
+    );
+    const promises = [];
+    let files: File[] = [];
+
+    if (!Array.isArray(filesOrFolder)) {
+      const directoryFiles = await this.bucket.getFiles({
+        prefix: filesOrFolder,
+      });
+      files = directoryFiles[0];
+    } else {
+      files = filesOrFolder.map(curFile => {
+        if (typeof curFile === 'string') {
+          return this.bucket.file(curFile);
+        }
+        return curFile;
+      });
+    }
+
+    const stripRegexString = options.stripPrefix
+      ? `^${options.stripPrefix}`
+      : EMPTY_REGEX;
+    const regex = new RegExp(stripRegexString, 'g');
+
+    for (const file of files) {
+      const passThroughOptionsCopy = extend(
+        true,
+        {},
+        options.passthroughOptions
+      );
+      if (options.prefix) {
+        passThroughOptionsCopy.destination = path.join(
+          options.prefix || '',
+          passThroughOptionsCopy.destination || '',
+          file.name
+        );
+      }
+      if (options.stripPrefix) {
+        passThroughOptionsCopy.destination = file.name.replace(regex, '');
+      }
+      promises.push(limit(() => file.download(passThroughOptionsCopy)));
+    }
+
+    return Promise.all(promises);
+  }
+
+  /**
+   * @typedef {object} DownloadFileInChunksOptions
+   * @property {number} [concurrencyLimit] The number of concurrently executing promises
+   * to use when downloading the file.
+   * @property {number} [chunkSizeBytes] The size in bytes of each chunk to be downloaded.
+   * @experimental
+   */
+  /**
+   * Download a large file in chunks utilizing parallel download operations. This is a convenience method
+   * that utilizes {@link File#download} to perform the download.
+   *
+   * @param {object} [file | string] {@link File} to download.
+   * @param {DownloadFileInChunksOptions} [options] Configuration options.
+   * @returns {Promise<DownloadResponse>}
+   *
+   * @example
+   * ```
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('my-bucket');
+   * const transferManager = new TransferManager(bucket);
+   *
+   * //-
+   * // Download a large file in chunks utilizing parallel operations.
+   * //-
+   * const response = await transferManager.downloadLargeFile(bucket.file('large-file.txt');
+   * // Your local directory now contains:
+   * // - "large-file.txt" (with the contents from my-bucket.large-file.txt)
+   * ```
+   * @experimental
+   */
+  async downloadFileInChunks(
+    fileOrName: File | string,
+    options: DownloadFileInChunksOptions = {}
+  ): Promise<void | DownloadResponse> {
```

Comment:
What are your thoughts on data validation with this method? Since `file.download` with `{start, end}` skips validation, we may want to add support or mention it in the description.

Reply:
Good question... I will definitely add a callout in the comments that this will skip validation. Do we currently have a means of validating a file after it has been downloaded? I know you did a bunch of work in this area but I thought it only worked with streamed input?

Reply:
I believe it's technically possible to CRC32C chunks and then "sum" them into the final checksum, such that you could do this kind of chunked operation and still checksum without an additional byte scan. If you need MD5 or can't do the previous algorithm, I think validation would require a full read of the completed file.

Reply:
> Do we currently have a means of validating a file after it has been downloaded?

Yep (although we may be able to do this as we're receiving each chunk):
https://github.com/googleapis/nodejs-storage/blob/a61e6194c50fa2fb8d22d37626a62c5352275850/src/crc32c.ts#L257

> I know you did a bunch of work in this area but I thought it only worked with streamed input?

It's possible to use `CRC32C` with/out streams (`HashStreamValidator` is the class that wraps `CRC32C` for streams):
https://github.com/googleapis/nodejs-storage/blob/a61e6194c50fa2fb8d22d37626a62c5352275850/src/crc32c.ts#L134
https://github.com/googleapis/nodejs-storage/blob/a61e6194c50fa2fb8d22d37626a62c5352275850/src/hash-stream-validator.ts#L30-L34

Reply:
Are we comfortable with only allowing CRC32C. In my mind something is better than nothing... I can take a look at implementing it if we feel strongly this should be included.